src/connections.tsx

```diff
@@ -187,6 +187,59 @@ export function moveRelations(
   };
 }
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function getSharesFromPublicKey(publicKey: PublicKey): number {
+  return 10000; // TODO: implement
+}
+
+function filterVoteRelation(
+  relations: List<Relations>,
+  head: ID,
+  type: ID
+): List<Relations> {
+  const filteredRelations = relations.filter((relation) => {
+    return shortID(relation.head) === shortID(head) && relation.type === type;
+  });
+
+  const latestRelationsByAuthor = filteredRelations.reduce((acc, relation) => {
+    const isFound = acc.get(relation.author);
+    if (!!isFound && isFound.updated > relation.updated) {
+      return acc;
+    }
+    return acc.set(relation.author, relation);
+  }, Map<PublicKey, Relations>());
+  return latestRelationsByAuthor.toList();
+}
+
+export function countRelationVotes(
```

Comment:
I would prefer to have it more abstract, so the algorithm is more clear.  Parameters should just be lists of strings for the voting algorithm, everything else like getting relations etc. should just wrap around.

function aggregateVotes(votes: List<List<string>>, weights: List<number>) => List<string> // number of weights needs to equal number of Lists

Reply:
ok, so only lists of list<items>, i'll refactor